cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -1014,21 +1008,9 @@ void getServiceInstances(HttpResponder responder,
       }
 
       // If the runnable name is the same as the service name, then uses the service spec, otherwise use the worker spec
-      int instances;
-      if (specification.getName().equals(runnableName)) {
-        instances = specification.getInstances();
-      } else {
-        ServiceWorkerSpecification workerSpec = specification.getWorkers().get(runnableName);
-        if (workerSpec == null) {
-          responder.sendStatus(HttpResponseStatus.NOT_FOUND);
-          return;
-        }
-        instances = workerSpec.getInstances();
-      }
-
-      responder.sendJson(HttpResponseStatus.OK,
-                         new ServiceInstances(instances, getInstanceCount(namespaceId, appId, ProgramType.SERVICE,
-                                                                          serviceId, runnableName)));
+      int instances = specification.getInstances();
```

Comment:
Does this affect documentation? Do we have an issue tracking this: both the deletion and the "there is no service worker support in v3 APIs."?